osidb/middleware.py

```diff
@@ -32,3 +34,41 @@ def __call__(self, request):
             _logger.info(f"ACLs set from middleware for user {request.user.username}")
             set_user_acls(request.user.groups.all())
         return self.get_response(request)
+
+
+def OsidbHistoryMiddleware(get_response):
+    """
+    Provides pghistory context with user email or name for use in
+    FlawAudit model, overwriting the default behaivor of pghistory
+    HistoryMiddleware to provide an integer user id (pk)
+    """
+
+    def _setattr(self, attr, value):
+        _value = value
+        if attr == "user":
+            if getattr(value, "is_authenticated", False):
+                if getattr(value, "email", False):
+                    _value = value.email
+                if getattr(value, "username", False):
+                    _value = value.username
```

Comment:
In this case `username` has higher preference over `email` because of two separate `if` but in the case bellow in the `middleware` method, the `email` has higher preference over `username` because of `if elif`, shouldn't these two match?

Reply:
Missed this, thank you!